### PR TITLE
[Optimization] Support logprob overlap in speculative decoding

### PR DESCRIPTION
**Pull request by Deleter-D**
## Motivation

Improve performance of speculative decoding by overlapping logprob computation with token acceptance operations.

## Modifications

- Merged `speculate_get_target_logits` into `speculate_get_accept_tokens_and_logits` to enable kernel-level overlap
- Added `compute_cu_batch_offset_kernel` for efficient batch offset calculation on GPU
- Integrated accept tokens retrieval with target logits extraction in a single kernel
- Removed Python-side `cu_batch_token_offset` computation; now computed on GPU via `compute_cu_batch_offset_kernel`
- Optimized D2H copy of `cu_batch_token_offset` using pinned memory for async transfer

## Usage or Command

N/A

## Accuracy Tests

N/A

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/model_executor/layers/sample/logprobs.py [hunk 1]
```diff
@@ -184,43 +184,27 @@ def build_output_logprobs(
...
-            "int32"
-        )
-        cu_batch_token_offset = paddle.concat(
-            [paddle.to_tensor([0]), paddle.cumsum(share_inputs["accept_num"][:max_occupied_slots])]
-        ).astype("int32")
-        share_inputs["cu_batch_token_offset"] = cu_batch_token_offset
-
         output_logits = paddle.empty(
-            [share_inputs["accept_num"][:max_occupied_slots].sum(), logits.shape[1]],
+            [real_bsz * max_draft_token_num_plus_1, logits.shape[1]],
```
[the remark this answers is not in the document]
Comment (Deleter-D) [reply fastdeploy/model_executor/layers/sample/logprobs.py]: save时有兜底

File: fastdeploy/model_executor/layers/sample/logprobs.py [hunk 2]
```diff
@@ -231,4 +215,4 @@ def build_output_logprobs(
 
     logprobs_tensors = gather_logprobs(raw_logprobs, num_logprobs, token_ids=token_ids)
```
[the remark this answers is not in the document]
Comment (Deleter-D) [reply fastdeploy/model_executor/layers/sample/logprobs.py:216]: save时会有兜底

**Review by Sunny-bot1 — DISMISSED** [review]

**Comment by huicongyao** [comment]
LGTM

**Review by freeliuzc — DISMISSED** [review]

**Review by freeliuzc — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by yongqiangma — APPROVED** [review]
LGTM
